pilot/pkg/mcp/coredatamodel.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package coredatamodel
+
+import (
+	"fmt"
+
+	mcpclient "istio.io/istio/galley/pkg/mcp/client"
+	"istio.io/istio/pilot/pkg/model"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Updater struct {
+	CoreDataModel
+	configStore      model.ConfigStore
+	configDescriptor model.ConfigDescriptor
+	descriptors      map[string]model.ProtoSchema
+}
+
+type CoreDataModel interface {
+	model.ConfigStore
+	// TODO: ServiceDiscovery
+}
+
+func NewUpdater(store model.ConfigStore, configDescriptor model.ConfigDescriptor) *Updater {
+	descriptors := make(map[string]model.ProtoSchema, len(configDescriptor))
+	for _, config := range configDescriptor {
+		descriptors[config.MessageName] = config
+	}
+
+	return &Updater{
+		configStore:      store,
+		configDescriptor: configDescriptor,
+		descriptors:      descriptors,
+	}
+}
+
+func (u *Updater) Update(change *mcpclient.Change) error {
+	for _, obj := range change.Objects {
+		descriptor, ok := u.descriptors[change.MessageName]
+		if !ok {
+			return fmt.Errorf("unknown type: %s received by updater", change.MessageName)
+		}
+		c, exists := u.configStore.Get(descriptor.Type, obj.Metadata.Name, "")
+		if exists {
+			c.Spec = obj.Resource
+			_, err := u.configStore.Update(*c)
+			return err
+		}
+		config := model.Config{
+			ConfigMeta: model.ConfigMeta{
+				Type:              descriptor.Type,
+				Group:             descriptor.Group,
+				Version:           descriptor.Version,
+				Name:              obj.Metadata.Name,
+				Namespace:         "",
+				Domain:            "",
+				Labels:            map[string]string{},
+				Annotations:       map[string]string{},
+				ResourceVersion:   "",
+				CreationTimestamp: meta_v1.Time{},
+			},
+			Spec: obj.Resource,
+		}
+		_, err := u.configStore.Create(config)
```

Comment:
Please don't do this - we don't want to create resources. Not sure if k8s adapter actually creates , but that's the semantic.

Reply:
what do you mean by "create" here?
we expect to populate the store with resources so that List() can return resources to the proxy serving layer. 

are you referring to the use of the public `Create` in the controller interface?